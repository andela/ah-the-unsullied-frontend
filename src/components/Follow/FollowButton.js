import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { follow } from '../../actions/profile/followActions';

export class FollowButton extends Component {
  followBtn = () => {
    const profile = this.props;
    const followStatus = profile.following;
    const auth = this.props.auth;
    if (auth.isAuthenticated) {
      const profileUser = profile.username;
      const username = auth.user.username;
      if (profileUser !== username) {
        const fbuttn = followStatus ? (
          <div>
            <button onClick={this.followProfile} className="follow">
              UnFollow
            </button>
          </div>
        ) : (
          <div>
            <button onClick={this.followProfile} className="follow">
              Follow
            </button>
          </div>
        );
        return fbuttn;
      }
    }
  };

  followProfile = () => {
    const { username } = this.props;
    this.props.follow(username);
  };

  render() {
    return <div>{this.followBtn()}</div>;
  }
}

FollowButton.propTypes = {
  auth: PropTypes.object.isRequired,
  follow: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired
};

export const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { follow }
)(FollowButton);
