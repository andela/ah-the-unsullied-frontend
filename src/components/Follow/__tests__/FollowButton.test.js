import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';
import {FollowButton, mapStateToProps} from '../FollowButton';


Enzyme.configure({ adapter: new Adapter() });
describe('follow button', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  beforeEach(() => {
    props = {
      bio:'i have abio', 
      username:'nesh', 
      image:'hshhshs',

      auth:{
        isAuthenticated:true,
        user:{
          username:'nesh',
        }
      },

      follow: jest.fn(() => {
        Promise.resolve();
      }),
     
    };

    state = {
      auth:{

      },
      
    };
    wrapper = shallow(<FollowButton {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should render the follow button', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the state when calling handLike', () => {
    const username = 'nesh';
    wrapperInstance.followProfile();
    expect(props.follow).toHaveBeenCalledWith(username);
  });


  it('mapStateToProps', () => {
    const propsState = mapStateToProps(state);
    expect(propsState.auth).toEqual(state.auth);
  });
});