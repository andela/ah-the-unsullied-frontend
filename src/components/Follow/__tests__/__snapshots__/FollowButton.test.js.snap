// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`follow button should render the follow button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FollowButton
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      }
    }
    bio="i have abio"
    follow={[MockFunction]}
    image="hshhshs"
    username="nesh"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
